core/src/main/java/org/mule/api/transport/NonBlockingResponseReplyToHandler.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.transport;
+
+import org.mule.api.MessagingException;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.MuleMessage;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.execution.MessageProcessorExecutionTemplate;
+import org.mule.api.processor.CompletionHandler;
+import org.mule.processor.chain.NonBlockingProcessorIterator;
+import org.mule.processor.chain.ProcessorIterator;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * {@link org.mule.api.transport.NonBlockingResponseReplyToHandler} is used to process the Flow response phase when the
```

Comment:
"{@link org.mule.api.transport.NonBlockingResponseReplyToHandler} is used to process " -> Processes